### DISCUSSION TITLE
Searching Using Like Clause Gives An Error

### DISCUSSION THREAD

**Discussion by Elvis254** [codeigniter4/shield#1048] [Q&A] [answered]
### PHP Version

8.2.0

### CodeIgniter4 Version

4.4.5

### Shield Version

1.0.1

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### Database

MariaDB 10.2

### Did you customize Shield?

I customized the `UserModel` class to include a function to generate a full name of a user from their first and last name before insert and before update. I also customized the `User` entity class to include two functions to set the first and last name of a user after cleaning the string input and performing formatting on it like capitalization.

### What happened?

Searching using the `like` clause doesn't work. Here is the code am using: `auth()->getProvider()->like(['username' => $query, 'secret' => $query])->orderBy('id', 'ASC')->paginate(100, 'usersList', null, 4)`

### Steps to Reproduce

Include `'secret'` column in the `where` clause so that someone can use the `like` clause to search for a user using both the username and email address.

### Expected Output

The system should display a list of matches or no matches based on the provided query string.

**Comment by kenjis** [comment]
There is no `secret` column in `users` table:
https://github.com/codeigniter4/shield/blob/0802f39638c2b6aa10ad7bfaddef3fb00f2ae126/src/Database/Migrations/2020-12-28-223112_create_auth_tables.php#L48-L56

See https://shield.codeigniter.com/getting_started/concepts/#user-identities

**Reply by Elvis254** [reply]
![Untitled](https://github.com/codeigniter4/shield/assets/5276840/ac33f20e-cc4a-4f7e-aef9-e1cad0fa6a88)

That is the error am getting when I write `auth()->getProvider()->like(['username' => $query, 'secret' => $query])->orderBy('id', 'ASC')->paginate(100, 'usersList', null, 4)` as shown above. Are we able to use the two fields when searching for a user because the provider fetches values from both the `users` and `identities` tables when listing? This code `auth()->getProvider()->orderBy('id', 'ASC')->paginate(100, 'usersList', null, 2)` works for listing purposes no need to write a `join` clause to get the `secret` value from the `identities` table.

**Reply by kenjis** [reply, marked as answer]
> The CodeIgniter’s Model provides convenience features and additional functionality that people commonly use to make working with a **single table** in your database more convenient.
https://codeigniter4.github.io/CodeIgniter4/models/model.html#models

You get the error because there is no column named `secret` in the table.
So you need to `join` by yourself.

**Reply by Elvis254** [reply]
Okay let me do it.
